Source/WebKit/UIProcess/ios/WKScrollView.mm

```diff
@@ -464,6 +470,25 @@ - (void)_updateScrollability
     [super setScrollEnabled:(_scrollEnabledByClient && _scrollEnabledInternal)];
 }
 
+- (void)setBounces:(BOOL)value
+{
+    _bouncesSetByClient = value;
+    [self _updateScrollability];
+}
+
+- (void)_setBouncesInternal:(BOOL)horizontal vertical:(BOOL)vertical
+{
+    _bouncesHorizontalInternal = horizontal;
+    _bouncesVerticalInternal = vertical;
+    [self _updateBouncability];
+}
+
+- (void)_updateBouncability
+{
+    [super setBouncesHorizontally:(_bouncesSetByClient && _bouncesHorizontalInternal)];
```

Comment:
This means that a client can only turn a bounces=YES into a bounces=NO, but should they be allowed to enable bouncing on a page with `overscroll-behavior:none`? I don't know. But since you made a decision here, document it in the commit message.

Reply:
Added to the commit message. I agree that it's unclear what the right behavior here should be. For the follow up patch I'll try and see what other properties are doing and make them consistent.